src/gt4py/next/program_processors/runners/dace_fieldview/gtir_builtin_translators.py

```diff
@@ -366,36 +366,36 @@ def translate_as_fieldop(
     """
     assert isinstance(node, gtir.FunCall)
     assert cpm.is_call_to(node.fun, "as_fieldop")
-    assert isinstance(node.type, ts.FieldType)
 
     fun_node = node.fun
     assert len(fun_node.args) == 2
-    stencil_expr, domain_expr = fun_node.args
+    fieldop_expr, domain_expr = fun_node.args
```

Comment:
The only thing that you changed here is that you renamed `stencil_expr` to `fieldop_expr` and then later you essentially alias it back to to `stencil_expr`, but handle a specific case differently.

I could imagine, that it would look better and cleaner, if you would move the unpacking into the `if`.
I know it will leads to a little bit of code duplication, but I think that it is worth it, since you get rid of `fieldop_expr` which does not serve any purpose.